### PR TITLE
Nick dev

### PR DESCRIPTION
**Pull request by nickl425**
idk if we want to merge since i changed cpics.py logic to include size filtering and add labels. I think the original cpics only worked without labels anyways. But if you want to convert data into train and test use this from now on

### CODE REVIEW

File: dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py [hunk 1]
```diff
@@ -48,40 +51,62 @@ def find_files(folder,  ext=None):
 
 
 def main(args):
-    if not args.extension.startswith('.'): args.extension = '.' + args.extension
+    if not args.extension.startswith('.'):
+        args.extension = '.' + args.extension
```
Comment (Storyxx) [inline dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py:55]: 🤩

File: dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py [hunk 2]
```diff
@@ -48,40 +51,62 @@ def find_files(folder,  ext=None):
...
+
+    lmdb_imgs_path = os.path.join(lmdb_dir , f"{args.dataset_name}_imgs")
+    lmdb_labels_path = os.path.join(lmdb_dir, f"{args.dataset_name}_labels")
     os.makedirs(lmdb_imgs_path, exist_ok=True)
-    # os.makedirs(lmdb_labels_path, exist_ok=True)
+    os.makedirs(lmdb_labels_path, exist_ok=True)
 
     env_imgs = lmdb.open(lmdb_imgs_path, map_size=MAP_SIZE_IMG)
-    # env_labels = lmdb.open(lmdb_labels_path, map_size=MAP_SIZE_IMG)
+    env_labels = lmdb.open(lmdb_labels_path, map_size=MAP_SIZE_META)
```
Comment (Storyxx) [inline dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py:71]: Why is it called `MAP_SIZE_META` when it sets the memory for the label lmdb database. `MAP_SIZE_LABEL` would be better.

File: dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py [hunk 3]
```diff
@@ -48,40 +51,62 @@ def find_files(folder,  ext=None):
...
     ):
-        for img_idx, img in tqdm(enumerate(sorted(img_abspath)), total=len(img_abspath)):
-            imgname = img_relpath[img_idx]
-            img_name_cleaned = "".join(e for e in imgname if e.isalnum() or e == '_')
-            do_print = img_idx % 50 == 0
-            if do_print:
-                print(f'idx: {img_idx}/{len(img_abspath)}, fov: "{img_name_cleaned}"')
+        for img_idx, (rel_path, abs_path) in tqdm(enumerate(zip(img_relpath, img_abspath)), total=len(img_abspath)):
+            #if img_idx >= 10:  # testing
+            #    break
```
Comment (Storyxx) [inline dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py:79]: Maybe clean up test code ^^

File: dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py [hunk 4]
```diff
@@ -48,40 +51,62 @@ def find_files(folder,  ext=None):
...
     env_imgs.close()
-    #env_labels.close()
+    env_labels.close()
+    print(f"Finished importing from {args.dataset_path} and subdirectories, saved at: {lmdb_imgs_path}")
+
+
+
+    env_imgs.close()
+    env_labels.close()
     print(f"Finished importing from {args.dataset_path} and subdirectories, saved at: {lmdb_imgs_path}")
```
Comment (Storyxx) [inline dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py]: Are you sure you need this twice?

File: dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py [hunk 5]
```diff
@@ -106,3 +140,5 @@ def get_args_parser():
     args_parser = get_args_parser()
     args = args_parser.parse_args()
     sys.exit(main(args))
+
+# example usage: !python save_cpics_pngs_to_lmdb.py --dataset_path="/home/nick/Downloads/113201/FlowCamNet/imgs" --lmdb_dir_name="/home/nick/Documents/ws24/lmdb/bigger_imgs/" --min_size=128 --dataset_name="FlowCamNet"
```
Comment (Storyxx) [inline dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py:144]: Please consider moving this example to the top of the file and leaving a blank line at the bottom (for GitHub)

File: dinov2/data/datasets/extended.py [hunk 6]
```diff
@@ -27,17 +27,27 @@ def get_target(self, index: int) -> Any:
         raise NotImplementedError
 
     def __getitem__(self, index: int) -> Union[Tuple[Any, Any], torch.Tensor, Image.Image]:
-        num_channels = 3  # base number
+        num_channels = 1  # base number
         img_bytes = self.get_image_data(index)
-        if isinstance(img_bytes, list):  # image
-            image = [torch.from_numpy(iio.imread(ch_bytes, index=None)) for ch_bytes in img_bytes]
-            image = torch.stack(image, dim=0)
+        if True:  # image
```
Comment (Storyxx) [inline dinov2/data/datasets/extended.py:32]: hmmmm

File: dinov2/data/datasets/extended.py [hunk 7]
```diff
@@ -27,17 +27,27 @@ def get_target(self, index: int) -> Any:
...
+            image = torch.frombuffer(np.copy(img_bytes), dtype=torch.uint8)
+        #    print(image.shape)
+            image = decode_image(image)
             image = (image / 255.0).to(torch.float32)
+        #    # image = torch.stack(image, dim=0)
+        #     print(image.shape[0])
+        #     image = (image / 255.0).to(torch.float32)
+        #     image = image.reshape(num_channels, image.shape[0], image.shape[1])
+        #    print(image.shape)
```
Comment (Storyxx) [inline dinov2/data/datasets/extended.py:41]: remove?

File: dinov2/data/datasets/extended.py [hunk 8]
```diff
@@ -27,17 +27,27 @@ def get_target(self, index: int) -> Any:
...
+            
         else:
             try:
                 # have to copy bc stream not writeable
                 image = torch.frombuffer(np.copy(img_bytes), dtype=torch.uint8)
-
+                print("i suck")
```
Comment (Storyxx) [inline dinov2/data/datasets/extended.py:47]: Don't be mean to yourself please! <3

File: dinov2/data/datasets/extended.py [hunk 9]
```diff
@@ -27,17 +27,27 @@ def get_target(self, index: int) -> Any:
...
+                print(image.shape[0])
+                print(num_channels)
+                print(image.shape[1])
                 if "plankton" in str(self.root):
```
Comment (Storyxx) [inline dinov2/data/datasets/extended.py:51]: Oh look it's the `plankton` catch 🤣

File: dinov2/data/datasets/extended.py [hunk 10]
```diff
@@ -56,11 +66,18 @@ def __getitem__(self, index: int) -> Union[Tuple[Any, Any], torch.Tensor, Image.
                     raise RuntimeError(f"can not read image for sample {index}") from e
 
         target = self.get_target(index)
-        target = TargetDecoder(target).decode()
-
+        #print("target:" + str(target))
+        #print("image:" + str(image.shape))
+        #print(self.transforms)
+        intiala, initialb = image.shape[1], image.shape[2]
```
Comment (Storyxx) [inline dinov2/data/datasets/extended.py:72]: This is weirdly named (why not width, height) and unused

File: dinov2/data/datasets/lmdb_dataset.py [hunk 11]
```diff
@@ -39,9 +40,14 @@ def get_target(self, index: int) -> Optional[Target]:
             return None
         else:
             entries = self._get_entries()
-            if self.with_targets:
-                class_index = entries[index]["class_id"]
-                return int(class_index)
+            if True:#if self.with_targets:
```
Comment (Storyxx) [inline dinov2/data/datasets/lmdb_dataset.py:43]: hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

File: dinov2/data/transforms.py [hunk 12]
```diff
@@ -8,7 +8,7 @@
 import torch
 from kornia import augmentation
 from torchvision.transforms import v2
-
+from PIL import Image
```
Comment (Storyxx) [inline dinov2/data/transforms.py:11]: This seems to be unused

File: dinov2/eval/knn.py [hunk 13]
```diff
@@ -412,8 +414,8 @@ def eval_knn(
 def eval_knn_with_model(
     model,
     output_dir,
-    train_dataset_str="ImageNet:split=TRAIN",
-    val_dataset_str="ImageNet:split=VAL",
+    train_dataset_str="LMDB:split=TRAIN",
+    val_dataset_str="LMDB:split=VAL",
```
Comment (Storyxx) [inline dinov2/eval/knn.py:418]: Nice :)

**Review by Storyxx — COMMENTED** [review]
You probably could merge, we will fix it later :)

File: dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py (hunk 4 above)
Comment (nickl425) [reply dinov2/data/dataset_creation/save_cpics_pngs_to_lmdb.py]: yes :)
